enhancements/20220530-vxlan-enhancement.md

```diff
@@ -0,0 +1,521 @@
+
+With following static added entries, the gratuitous ARP (Broadcast) is head-end replicated to all VTEPs. Normally all BUM frame will be duplicated to all these destinations.
+
+It is not so efficiency, but kube-vip also does not floate the IP so frequently in most time.
+
+```
+in NODE2:
+bridge fdb append 00:00:00:00:00:00 dev vxlan100 dst 192.168.5.3
+bridge fdb append 00:00:00:00:00:00 dev vxlan100 dst 192.168.5.4
+
+in NODE3:
+bridge fdb append 00:00:00:00:00:00 dev vxlan100 dst 192.168.5.2
+bridge fdb append 00:00:00:00:00:00 dev vxlan100 dst 192.168.5.4
+
+in NODE4:
+bridge fdb append 00:00:00:00:00:00 dev vxlan100 dst 192.168.5.2
+bridge fdb append 00:00:00:00:00:00 dev vxlan100 dst 192.168.5.3
+```
+
+#### Story 10
+
+Multi-instances of VXLAN
+
+As Host Harvester Cluster can have multi isolated VLAN network, it is possible to have multi-instances of VXLAN on top of those isolated VLAN Network.
+
+Challenges:
+
+1. Overlay IP allocation, can they overlap ?
+
+2. Is it allowed of cross VXLAN instance communication ?
+
+
+### User Experience In Detail
+
+#### An example of VXLAN user operation
+
+To now, the user will be able to use Harvester VXLAN network to create isolated Guest Kubernetes Cluster with following steps.
+
+```
+(1) Install Host Harvester Cluster, wait for it's ready
+(2) Create VLAN network on second NIC, say VLAN 5, IP DHCP from provider network, CIDR e.g. 192.168.5.0/24
+(3) Create VXLAN network on top of VLAN, say VXLAN 1, based on VLAN 5,  set CIDR, e.g. 10.0.0.0/8
+     VXLAN Local IP and remote peers in each NODE are fetched from APIServer.
+     Related VXLAN L3 Gateway and NAT are also deployed.
+
+(4) Create a VNI, say 100, set CIDR, e.g. 10.99.100.0/24
+     VXLAN 1, VNI 100, the VXLAN tunnel in each NODE are established, each node has a VTEP instance, say vxlan_1_100
+(5) Create a group of VMs, say vmg_1, the CNI is binded to VNI 100
+     each vxlan_1_100 will have related forwarding db (MAC address of VM, local port/remote VTEP IP)
+     selecting head end replication
+     add MAC 00.00.00.00.00.00 for each remote peer, for BUM packet processing
+     VM--VM communication is ready
+(6) Provision a guest k8s cluster, cluster_1 in vmg_1
+(7) Add a LB/VIP for cluster_1 (10.99.100.99), a corresponding VIP is allocated in provider network, 192.168.5.99
+     cluster_1 is able to communicate with provider network
+
+
+repeat (4)~(7) to create more Guest Kubernetes Clusters.
+
+```
+
+[back to user stories](#user-stories)
+
+#### Harvester network extension proposal
+
+A proposal, for discussion.
+
+.
+
+### API changes
+
+## Design
+
+### Implementation Overview
+
+Overview on how the enhancement will be implemented.
+
+An analysis of Linux VXLAN implementation.
+
+Noteablely, at the view of VXLAN, the linux bridge instance is a part of it's functionality, it should be automatically created/deleted with VXLAN.
```

Comment:
Not always.
Vxlan interface could be standalone if you used it as how Flannel used it.